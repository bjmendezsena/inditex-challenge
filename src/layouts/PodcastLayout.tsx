import { Outlet } from "react-router-dom";

export const PodcastLayout = () => {
  return (
    <div>
      <h1>Podcast Layout</h1>
      <Outlet />
    </div>
  );
};
