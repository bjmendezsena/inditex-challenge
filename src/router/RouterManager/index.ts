export * from './RouteName';
export * from './RoutePath';
