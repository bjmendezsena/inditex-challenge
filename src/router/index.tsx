import { RouterProvider, createBrowserRouter } from "react-router-dom";
import routes from "./Routes";

export const router = createBrowserRouter(routes);

const AppRouter = () => <RouterProvider router={router} />;

export default AppRouter;
