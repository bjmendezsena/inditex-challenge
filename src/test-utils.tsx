import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { theme } from "./theme";

export const testQueryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      refetchOnMount: true,
      cacheTime: 0,
    },
  },
  logger: {
    log: console.log,
    warn: console.warn,
    // âœ… no more errors on the console for tests
    error: process.env.NODE_ENV === "test" ? () => {} : console.error,
  },
});

export const queryWrapper = ({ children }: { children: React.ReactNode }) => (
  <QueryClientProvider client={testQueryClient}>{children}</QueryClientProvider>
);

const DefaultTheme = extendTheme(theme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
