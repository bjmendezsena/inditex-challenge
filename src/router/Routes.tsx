import { RouteObject, Navigate } from "react-router-dom";
import * as Views from "../views";
import { MainLayout, PodcastLayout } from "../layouts";
import { RoutePath } from "./RouterManager";

const routes: RouteObject[] = [
  {
    element: <MainLayout />,
    children: [
      {
        path: RoutePath.PodcastList,
        element: <Views.PodcastListPage />,
      },
      {
        element: <PodcastLayout />,
        children: [
          {
            path: RoutePath.PodcastDetails,
            element: <Views.PodcastDetails />,
          },
          {
            path: RoutePath.EpisodeDetails,
            element: <Views.EpisodeDetails />,
          },
        ],
      },
      {
        path: "*",
        element: <Navigate to={RoutePath.PodcastList} replace />,
      },
    ],
  },
];

export default routes;
