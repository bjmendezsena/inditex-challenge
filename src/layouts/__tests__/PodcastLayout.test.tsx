import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { createMemoryHistory } from "history";
import { matchPath } from "react-router-dom";
import { render } from "../../test-utils";
import { PodcastLayout } from "../PodcastLayout";
import { RouterManager, RouteName } from "../../router";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    podcastId: "199",
  }),
}));

describe(`<${PodcastLayout.name}/>`, () => {
  const history = createMemoryHistory();
  const factoryComponent = () =>
    render(<PodcastLayout />, {
      history,
    });

  it("should render without crashing", () => {
    const { container } = factoryComponent();
    expect(container).toMatchSnapshot();
  });

  it("Should navigate to the podcast details page when clicking on the podcast card", () => {
    factoryComponent();

    userEvent.click(screen.getByTestId("podcast-details-card"));
    expect(
      matchPath(
        RouterManager.getUrl(RouteName.PodcastDetails, {
          podcastId: "199",
        }),
        history.location.pathname
      )
    ).toBeTruthy();
  });
});
