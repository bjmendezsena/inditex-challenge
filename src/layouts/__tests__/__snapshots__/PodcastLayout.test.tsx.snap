// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PodcastLayout/> should render without crashing 1`] = `
<div>
  <div
    class="css-9ywscx"
  >
    <div
      class="css-0"
    >
      <a
        href="/podcast/199"
      >
        <div
          class="chakra-card css-1h3y5ak"
          data-testid="podcast-details-card"
        >
          <div
            class="chakra-card__body css-bwepl0"
          >
            <div
              class="chakra-stack css-b93e9j"
            >
              <div
                class="chakra-skeleton css-fnhvzt"
              >
                <img
                  alt=""
                  class="chakra-image css-sitj8d"
                  src=""
                />
              </div>
              <div
                class="chakra-stack__divider css-701422"
              />
              <div
                class="css-1ofqig9"
              >
                <div
                  class="chakra-skeleton css-1mdjp85"
                />
                <div
                  class="chakra-skeleton css-1buwyc7"
                />
              </div>
              <div
                class="chakra-stack__divider css-701422"
              />
              <div
                class="css-1ofqig9"
              >
                <h2
                  class="chakra-heading css-18cb01y"
                >
                  Description
                </h2>
                <div
                  class="chakra-skeleton__group css-rltemf"
                >
                  <div
                    class="chakra-skeleton css-ctcxab"
                  />
                  <div
                    class="chakra-skeleton css-ctcxab"
                  />
                  <div
                    class="chakra-skeleton css-12r9lpq"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
    <div
      class="css-0"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
